// frost-react/dist/index.js
import { jsx as g } from "react/jsx-runtime";
import { useState as O, useEffect as x, useMemo as f } from "react";
function P() {
  const [e, n] = O(!1);
  return x(() => {
    if (typeof window > "u")
      return;
    const o = window.matchMedia("(max-width: 768px)");
    n(o.matches);
    const i = (s) => {
      n(s.matches);
    };
    return o.addEventListener ? (o.addEventListener("change", i), () => o.removeEventListener("change", i)) : (o.addListener(i), () => o.removeListener(i));
  }, []), e;
}
function a(e, n) {
  return Math.random() * (n - e) + e;
}
function d(e) {
  return Array.isArray(e) ? e : [e, e];
}
function X() {
  return typeof window > "u" ? !1 : window.matchMedia("(prefers-reduced-motion: reduce)").matches;
}
function Y() {
  if (typeof document > "u")
    return;
  const e = "snowfall-styles";
  if (document.getElementById(e))
    return;
  const n = document.createElement("style");
  n.id = e, n.textContent = `
@keyframes snowfall {
  0% {
    transform: translateY(-100vh) translateX(var(--snowfall-drift)) rotate(var(--snowfall-rotation-start));
    opacity: var(--snowfall-opacity);
  }
  100% {
    transform: translateY(100vh) translateX(calc(var(--snowfall-drift) * 2)) rotate(var(--snowfall-rotation-end));
    opacity: calc(var(--snowfall-opacity) * 0.3);
  }
}

.snowfall-container {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  z-index: var(--snowfall-z-index, 50);
  overflow: hidden;
}

.snowfall-flake {
  position: absolute;
  top: -10px;
  left: var(--snowfall-left);
  font-size: var(--snowfall-size);
  color: var(--snowfall-color, #ffffff);
  opacity: var(--snowfall-opacity);
  animation: snowfall var(--snowfall-duration) linear infinite;
  animation-delay: var(--snowfall-delay);
  filter: drop-shadow(0 0 2px rgba(255, 255, 255, 0.5));
  user-select: none;
  will-change: transform;
}

@media (prefers-reduced-motion: reduce) {
  .snowfall-flake {
    animation: none;
  }
}
  `.trim(), document.head.appendChild(n);
}
const r = {
  enabled: !0,
  snowflakeCharacter: "❄️",
  speed: [6, 12],
  size: [12, 28],
  opacity: [0.5, 0.9],
  wind: [-20, 20],
  zIndex: 50,
  useMobileDetection: !0
};
function T({
  enabled: e = r.enabled,
  flakeCount: n,
  snowflakeCharacter: o = r.snowflakeCharacter,
  speed: i = r.speed,
  size: s = r.size,
  opacity: c = r.opacity,
  wind: w = r.wind,
  color: z,
  zIndex: S = r.zIndex,
  style: M,
  className: b,
  useMobileDetection: u = r.useMobileDetection
}) {
  const p = P(), E = f(() => typeof window > "u" ? !1 : X(), []);
  x(() => {
    Y();
  }, []);
  const m = f(() => n !== void 0 ? n : u && p ? 35 : 70, [n, u, p]), $ = f(() => {
    if (typeof window > "u")
      return [];
    const t = d(i), l = d(s), y = d(c), h = d(w);
    return Array.from({ length: m }, (_, I) => {
      const L = a(t[0], t[1]), A = a(l[0], l[1]), C = a(y[0], y[1]), k = a(h[0], h[1]), j = a(0, 100), D = a(0, 5), v = a(0, 360), F = v + a(-180, 180), N = 20 - L;
      return {
        id: I,
        left: j,
        size: A,
        opacity: C,
        wind: k,
        delay: D,
        duration: N,
        rotationStart: v,
        rotationEnd: F
      };
    });
  }, [m, i, s, c, w]);
  if (!e || E || typeof window > "u")
    return null;
  const R = {
    "--snowfall-z-index": S.toString(),
    ...M
  };
  return /* @__PURE__ */ g(
    "div",
    {
      className: `snowfall-container ${b || ""}`.trim(),
      style: R,
      "aria-hidden": "true",
      children: $.map((t) => {
        const l = {
          "--snowfall-left": `${t.left}%`,
          "--snowfall-size": `${t.size}px`,
          "--snowfall-opacity": t.opacity.toString(),
          "--snowfall-drift": `${t.wind}px`,
          "--snowfall-delay": `${t.delay}s`,
          "--snowfall-duration": `${t.duration}s`,
          "--snowfall-rotation-start": `${t.rotationStart}deg`,
          "--snowfall-rotation-end": `${t.rotationEnd}deg`,
          "--snowfall-color": z
        };
        return /* @__PURE__ */ g(
          "div",
          {
            className: "snowfall-flake",
            style: l,
            children: o
          },
          t.id
        );
      })
    }
  );
}
export {
  T as Snowfall
};
