// frost-react/dist/index.cjs
"use strict";Object.defineProperty(exports,Symbol.toStringTag,{value:"Module"});const g=require("react/jsx-runtime"),s=require("react");function q(){const[e,n]=s.useState(!1);return s.useEffect(()=>{if(typeof window>"u")return;const o=window.matchMedia("(max-width: 768px)");n(o.matches);const i=l=>{n(l.matches)};return o.addEventListener?(o.addEventListener("change",i),()=>o.removeEventListener("change",i)):(o.addListener(i),()=>o.removeListener(i))},[]),e}function a(e,n){return Math.random()*(n-e)+e}function f(e){return Array.isArray(e)?e:[e,e]}function D(){return typeof window>"u"?!1:window.matchMedia("(prefers-reduced-motion: reduce)").matches}function F(){if(typeof document>"u")return;const e="snowfall-styles";if(document.getElementById(e))return;const n=document.createElement("style");n.id=e,n.textContent=`
@keyframes snowfall {
  0% {
    transform: translateY(-100vh) translateX(var(--snowfall-drift)) rotate(var(--snowfall-rotation-start));
    opacity: var(--snowfall-opacity);
  }
  100% {
    transform: translateY(100vh) translateX(calc(var(--snowfall-drift) * 2)) rotate(var(--snowfall-rotation-end));
    opacity: calc(var(--snowfall-opacity) * 0.3);
  }
}

.snowfall-container {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  z-index: var(--snowfall-z-index, 50);
  overflow: hidden;
}

.snowfall-flake {
  position: absolute;
  top: -10px;
  left: var(--snowfall-left);
  font-size: var(--snowfall-size);
  color: var(--snowfall-color, #ffffff);
  opacity: var(--snowfall-opacity);
  animation: snowfall var(--snowfall-duration) linear infinite;
  animation-delay: var(--snowfall-delay);
  filter: drop-shadow(0 0 2px rgba(255, 255, 255, 0.5));
  user-select: none;
  will-change: transform;
}

@media (prefers-reduced-motion: reduce) {
  .snowfall-flake {
    animation: none;
  }
}
  `.trim(),document.head.appendChild(n)}const r={enabled:!0,snowflakeCharacter:"❄️",speed:[6,12],size:[12,28],opacity:[.5,.9],wind:[-20,20],zIndex:50,useMobileDetection:!0};function N({enabled:e=r.enabled,flakeCount:n,snowflakeCharacter:o=r.snowflakeCharacter,speed:i=r.speed,size:l=r.size,opacity:c=r.opacity,wind:u=r.wind,color:x,zIndex:S=r.zIndex,style:z,className:M,useMobileDetection:w=r.useMobileDetection}){const y=q(),b=s.useMemo(()=>typeof window>"u"?!1:D(),[]);s.useEffect(()=>{F()},[]);const p=s.useMemo(()=>n!==void 0?n:w&&y?35:70,[n,w,y]),E=s.useMemo(()=>{if(typeof window>"u")return[];const t=f(i),d=f(l),m=f(c),h=f(u);return Array.from({length:p},(T,$)=>{const I=a(t[0],t[1]),L=a(d[0],d[1]),j=a(m[0],m[1]),A=a(h[0],h[1]),C=a(0,100),k=a(0,5),v=a(0,360),O=v+a(-180,180),P=20-I;return{id:$,left:C,size:L,opacity:j,wind:A,delay:k,duration:P,rotationStart:v,rotationEnd:O}})},[p,i,l,c,u]);if(!e||b||typeof window>"u")return null;const R={"--snowfall-z-index":S.toString(),...z};return g.jsx("div",{className:`snowfall-container ${M||""}`.trim(),style:R,"aria-hidden":"true",children:E.map(t=>{const d={"--snowfall-left":`${t.left}%`,"--snowfall-size":`${t.size}px`,"--snowfall-opacity":t.opacity.toString(),"--snowfall-drift":`${t.wind}px`,"--snowfall-delay":`${t.delay}s`,"--snowfall-duration":`${t.duration}s`,"--snowfall-rotation-start":`${t.rotationStart}deg`,"--snowfall-rotation-end":`${t.rotationEnd}deg`,"--snowfall-color":x};return g.jsx("div",{className:"snowfall-flake",style:d,children:o},t.id)})})}exports.Snowfall=N;
